psm_diags.c

```diff
@@ -244,6 +245,7 @@ void *memcpy_check_one (memcpy_fn_t fn, void *dst, void *src, size_t n)
 	  ((uintptr_t) dst ^ (uintptr_t) src ^ (uintptr_t) n);
   unsigned int state;
   size_t i;
+  assert (n > 0);
```

Comment:
could you change this to use psmi_assert_always ?
Its somewhat of a wrapper we use around the linux assert.